app/server/appsmith-server/src/main/java/com/appsmith/server/configurations/SecurityConfig.java

```diff
@@ -176,6 +210,14 @@ public SecurityWebFilterChain securityWebFilterChain(ServerHttpSecurity http) {
                 .build();
     }
 
+    private boolean isAuthorizedToAccessInternal(String password) {
+        if (StringUtils.isNotEmpty(INTERNAL_PASSWORD)) {
+            return INTERNAL_PASSWORD.equals(password);
+        } else {
+            return true;
+        }
```

Comment:
Feel free to not agree with this suggestion. 🙂

```suggestion
        // Either configured password is empty, or it's equal to what we received.
        return StringUtils.isEmpty(INTERNAL_PASSWORD) || INTERNAL_PASSWORD.equals(password);
```